.ci/test_r_package.sh

```diff
@@ -130,6 +135,28 @@ elif [[ $R_BUILD_TYPE == "cran" ]]; then
 
     ./build-cran-package.sh || exit -1
 
+    if [[ "${TASK}" == "r-rchk" ]]; then
+        echo "Checking R package with rchk"
+        mkdir -p packages
+        cp ${PKG_TARBALL} packages
+        RCHK_LOG_FILE="rchk-logs.txt"
+        docker run \
+            -v $(pwd)/packages:/rchk/packages \
+            kalibera/rchk:latest \
```

Comment:
This docker image doesn't have any already configured R environment, right?

Reply:
I think that it does, but the way that it is set up, it expects that you've built the package outside and then passed it into a container running the image.

https://github.com/kalibera/rchk/blob/master/doc/DOCKER.md#checking-a-package-from-a-tarball

that's why I chose to use `docker run` here after our usual R setup stuff on an `ubuntu-latest` VM, instead of running a containerized job like

https://github.com/microsoft/LightGBM/blob/ec1debcee84e6b02f749282d9a501021370e74a9/.github/workflows/r_package.yml#L167-L171

I think it's ok since `docker` is available by default in the `ubuntu-latest` image: https://github.com/actions/virtual-environments/blob/7d0d3aa82ab8565cc43684bd08bf72fb6f5cd9f7/images/linux/Ubuntu2004-README.md

Reply:
Thanks! That's exactly what I wanted to clarify.

Reply:
Did you see that?
https://hub.docker.com/r/rhub/ubuntu-rchk
https://github.com/r-hub/rhub-linux-builders/blob/master/ubuntu-rchk/Dockerfile
Maybe this image can help us to run containerized job or we don't want to do that?..

Reply:
Oh interesting, no I didn't know about that. I can try it tonight or tomorrow.

Reply:
Alright I tested this tonight but couldn't quite get it working.

I tried the following from the root of the `LightGBM` repo.

```shell
# get a shell inside an ubuntu-rchk container
docker run \
    -v $(pwd):/opt/LightGBM \
    -w /opt/LightGBM \
    -it rhub/ubuntu-rchk \
    /bin/bash

# run the following inside the container
sh build-cran-package.sh
Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'Matrix'), repos = 'https://cran.r-project.org')"
rchk.sh lightgbm_*.tar.gz
```

Compilation of `{lightgbm}` and its dependencies succeeded, but running this failed with an error I don't understand.

Here is the tail of the logs:

> /usr/local/bin/wllvm++ -stdlib=libc++ -std=gnu++11 -shared -L/usr/local/lib -o lightgbm.so boosting/boosting.o boosting/gbdt.o boosting/gbdt_model_text.o boosting/gbdt_prediction.o boosting/prediction_early_stop.o io/bin.o io/config.o io/config_auto.o io/dataset.o io/dataset_loader.o io/file_io.o io/json11.o io/metadata.o io/parser.o io/train_share_states.o io/tree.o metric/dcg_calculator.o metric/metric.o objective/objective_function.o network/ifaddrs_patch.o network/linker_topo.o network/linkers_mpi.o network/linkers_socket.o network/network.o treelearner/data_parallel_tree_learner.o treelearner/feature_parallel_tree_learner.o treelearner/gpu_tree_learner.o treelearner/linear_tree_learner.o treelearner/serial_tree_learner.o treelearner/tree_learner.o treelearner/voting_parallel_tree_learner.o c_api.o lightgbm_R.o -pthread
installing to /home/docker/R-svn/packages/lib/00LOCK-lightgbm/00new/lightgbm/libs
** R
** data
** demo
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** testing if installed package can be loaded from temporary location
** checking absolute paths in shared objects and dynamic libraries
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
>* DONE (lightgbm)
/home/docker/rchk/scripts/check_package.sh: line 100:  2577 Killed                  $RCHK/src/$T $RBC $F > $FOUT 2>&1

So I think this PR should be merged in its current state. It would have been great to have a fully-containerized job so that we wouldn't have to spend CI time installing R for this, but I don't think investigating this further is the best use of our time, given that the current run time of the job is already not too bad (around 13 minutes).

Reply:
OK, got it! Thanks a lot for the try! If you have some time, I guess it worth to report this error to https://github.com/r-hub/rhub-linux-builders with example of successful run directly in https://github.com/kalibera/rchk.